### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by shyamd**
This PR adds documentation for Maggma as well as the build pipeline for documentation

- [x] docs build pipeline
- [x] basic builder documentation
- [x] advanced builder concepts
- [x] Using `MapBuilder`
- [x] API docs
- [x] Add descriptions for the Stores

### CODE REVIEW

File: src/docs/getting_started/stores.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+All `Store`s have a few basic arguments that are critical to understand. Any `Store` has two special fields: `key` and `last_updated_field`. The `key` defines how the `Store` tells documents part. Typically this is `_id` in MongoDB, but you could use your own field. `last_updated_field` tells `Store` how to order the documents by a date field.  `Store`s can also take a `Validator` object to make sure the data going into obeys some schema.
+
+### Using a Store
+
+Stores provide a number of basic methods that make easy to use:
+
+- query: standard mongo style `find` method that lets you search the store
+- query_one: same as above but limits to the first document
+- update: update the documents into the collection, this will override documents if the key field matches. You can temporarily provide extra fields to key these documents if you don't to maintain duplicates, for instance keying on both `key` and `last_udpated_field`
```
Comment (mkhorton) [inline src/docs/getting_started/stores.md]: typo

File: src/docs/getting_started/stores.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+- ensure_index: this creates an index the underlying data-source for fast querying
+- distinct: gets distinct values of a field
+- groupby: similar to query but performs a grouping operation and returns sets of documents
+- remove_docs: removes documents from the underlying data source
+- last_updated: finds the most recently updated `last_updated_field` value and returns that. Usefull for knowing how old a data-source is
+- newer_in: finds all documents that are newer in the target collection and returns their `key`s. This is a very usefull way of performing incremental processing
```
Comment (mkhorton) [inline src/docs/getting_started/stores.md]: typo, "usefull"

File: src/docs/getting_started/stores.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Using `Store`
```
Comment (mkhorton) [inline src/docs/getting_started/stores.md]: Maybe add a `## List of Stores` here? Could be super brief for now

File: src/docs/getting_started/advanced_builder.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
+# Advanced Builder Concepts
+
+There are a number of features in `maggma` designed to assist with advanced features:
+
```
Comment (mkhorton) [inline src/docs/getting_started/advanced_builder.md]: Maybe a list of advanced builders here also? With the MapBuilder one linking to the nice MapBuilder doc.

**Comment by mkhorton** [comment]
Looks great!

**Comment by shyamd** [comment]
Thanks! Once these tests finish, I'm gonna update the docs build to just update on merge for now and then merge.

**Pull request merged**